src/main/java/pl/cleankod/exchange/core/dto/MoneyDto.java

```diff
@@ -0,0 +1,8 @@
+package pl.cleankod.exchange.core.dto;
+
+public record MoneyDto(String amount, String currency) {
```

Comment:
You introduced DTOs to solve the serialization issues instead of extending the serialization engine. Nice! I like that. Let's discuss the trade-offs, though.